### DISCUSSION TITLE
Feature request: Branch protection - skip pull request requirement for specified paths

### DISCUSSION THREAD

**Discussion by audunsolemdal** [community/community#49181] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

My organization manages its infrastructure with terraform in a Github organization. The infrastructure for each application team has its own repository `wl-teamname`

Some structure examples:

```yaml
platform
  dev
    main.tf
  test
    main.tf
  production
    main.tf
.github/workflows/terraform-cicd.yaml
CODEOWNERS
```

We use the following branch protection settings for the main branch:
- Require a pull request before merging
- Require approvals (1)
- Require review from code owners

We only use the main branch and feature branches, where the main branch is the only protected branch. When we want a change in infrastructure a pull request is created towards the main branch, and a `terraform plan` action is executed. Once a PR is merged to the main branch, a workflow which runs `terraform apply` is run.

We combine the pull requests with the CODEOWNERS - so that members of the application teams can approve changes under platform/dev and platform/test folders. For changes to the platform/production folder, a code review from the central Cloud Platform team is required.

This works OK for the most part - but ideally we would like a new feature available. Basically, if the path is `platform/dev/xxx`, we would like to skip the requirement for an approved pull request by another member of the application team. This is intended so that changes in development can occur quickly without needing to involve other team members. In some cases there are no other team members who know terraform or IaC, making small changes a hassle.

I am aware that this could be solved by using a branch per environment, e.g. `dev` where we would not require a pull request, but we wish to keep our branching strategy. We also do not want to use Github environments and deployment gates.

**Comment by ervinb** [comment]
This is now supported [through rulesets](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/available-rules-for-rulesets#required-reviewers).

<img width="1167" height="597" alt="image" src="https://github.com/user-attachments/assets/820c9245-e892-4309-8833-759302839e40" />

Note that the global review requirement is set to 0, and the team-specific one has an all (`**`) matcher to require 1, and then ignores a specific path.
